ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala

```diff
@@ -397,4 +411,18 @@ object Endpoints {
   private def toJsValue[A: JsonWriter](a: A): Error \/ JsValue = {
     SprayJson.encode(a).liftErr(ServerError)
   }
+
+  private[http] final case class Forwarded(value: String) extends ModeledCustomHeader[Forwarded] {
+    override def companion = Forwarded
+    override def renderInRequests = true
+    override def renderInResponses = false
+    def proto: Option[String] =
+      Forwarded.re findFirstMatchIn value map (_.group(1).toLowerCase)
+  }
+
+  private[http] object Forwarded extends ModeledCustomHeaderCompanion[Forwarded] {
+    override val name = "Forwarded"
+    override def parse(value: String) = Try(new Forwarded(value))
+    private val re = raw"""(?i)proto\s*=\s*"?(https?)""".r
```

Comment:
This is pretty quick and dirty "parsing"; should we parse [rfc 7239 forwarded-elements](https://tools.ietf.org/html/rfc7239#section-4) instead? @leo-da

Reply:
I need to read on this first... but I expect Akka HTTP provides some utilities for this.

Reply:
> but I expect Akka HTTP provides some utilities for this

Not really. The closest they come is [`Connection`'s parser](https://github.com/akka/akka-http/blob/v10.1.11/akka-http-core/src/main/scala/akka/http/impl/model/parser/SimpleHeaders.scala#L77-L80), but that syntax is sufficiently incompatible to just crash on a Forwarded header. We can reuse [some of the parser components](https://github.com/akka/akka-http/blob/v10.1.11/akka-http-core/src/main/scala/akka/http/impl/model/parser/CommonRules.scala#L18-L63), though they are package-private and thus [not part of the stable API](https://doc.akka.io/docs/akka-http/10.1.11/common/http-model.html#parsing-rendering), but that _still_ does not include a forwarded-element parser anyway, so even odds we might as well define a new parser.

If what we want is a parser. That question thus reduces to whether avoiding the obvious edge cases in the quick and dirty approach is worth dealing with the above.

Reply:
I think this is fine. It's clearly possible to craft a Forwarded value that thwarts it, but if you're at the point of crafting the header you can already set it to whatever you want anyway.

The goal of this filter is not to be absolutely unbreakable, but to avoid accidental security holes.

Reply:
@garyverhaegen-da Makes sense to me, thanks.

Reply:
> We can reuse some of the parser components, though they are package-private and thus not part of the stable API
Nah... I thought it would be available. In any case I agree with @garyverhaegen-da it does not have to be bulletproof, it is just a fat finger check, a security advise to developers. If developer is crafting the header they supposed to understand the implications.